cumulusci/core/github.py

```diff
@@ -510,23 +511,16 @@ def get_oauth_scopes(response: Response) -> set:
     return authorized_scopes
 
 
-def catch_common_github_auth_errors(func: Callable) -> Callable:
-    """
-    A decorator catching the most common Github authentication errors.
-    """
-
-    @functools.wraps(func)
-    def inner(*args, **kwargs):
-        try:
-            return func(*args, **kwargs)
-        except (ResponseError, TransportError) as exc:
-            error_msg = format_github3_exception(exc)
-            if error_msg:
-                raise GithubApiError(error_msg)
-            else:
-                raise
-
-    return inner
+@contextmanager
+def catch_common_github_auth_errors(context: T.Any):
+    try:
+        yield
+    except (ResponseError, TransportError) as exc:
+        error_msg = format_github3_exception(exc)
+        if error_msg:
+            raise GithubApiError(f"{context}\n{error_msg}")
+        else:
+            raise
```

Comment:
What about an alternate approach?: Update `format_github3_exception` to append `f"\n\nURI: {exc.request.url}` to `user_warning`. This would give the requested resource in the body of the error message without requiring the caller to specify a context.

Reply:
```
(Pdb) exc.request.url
*** AttributeError: 'ForbiddenError' object has no attribute 'request'
(Pdb) dir(exc)
['__cause__', '__class__', '__context__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__suppress_context__', '__traceback__', '__weakref__', 'args', 'code', 'errors', 'message', 'msg', 'response', 'with_traceback']
(Pdb) vars(exc)
{'response': <Response [403]>, 'code': 403, 'errors': [], 'msg': '[No message]'}
(Pdb) exc.response.url
(Pdb) vars(exc.response)
{'_content': None, '_content_consumed': True, '_next': None, 'status_code': 403, 'headers': {'X-Github-Sso': 'partial-results; organizations=0810298,20348880'}, 'raw': None, 'url': None, 'encoding': None, 'history': [], 'reason': None, 'cookies': <RequestsCookieJar[]>, 'elapsed': datetime.timedelta(0), 'request': None}
```

Reply:
(I mean, we'd have to update the tests. 😄 )

Using it here:

https://github.com/SFDO-Tooling/CumulusCI/blob/7f797c9dbbf2cbfabaabf9530cd449c3742067da/cumulusci/core/github.py#L438-L440

_But_... looks like that `check_github_scopes` isn't getting tested.... so should put that on the todo list as well.

Reply:
Even in real-world use there is no URL. That's how I ended up down this path:

```
cci flow run dev_org --org dev
...
                    Resolving dependencies...                                                                                     
> /Users/pprescod/code/CumulusCI/cumulusci/core/github.py(398)format_github3_exception()
(Pdb) vars(exc)
{'response': <Response [404]>, 'code': 404, 'errors': [], 'msg': 'Not Found'}
```